internal/client/client_interceptor.go

```diff
@@ -136,9 +138,25 @@ func buildResourceControlInterceptor(
 					return nil, err
 				}
 				ruRuntimeStats.Update(consumption)
+				updateRUDetails(ctx, consumption, time.Duration(0))
 			}
 			return resp, err
 		}
 	}
 	return interceptor.NewRPCInterceptor("resource_control", interceptFn)
 }
+
+// updateRUDetails update delta ru consumption to global ExecDetails
+func updateRUDetails(ctx context.Context, consumption *rmpb.Consumption, waitDuration time.Duration) {
+	if consumption == nil {
+		return
+	}
+	if stmtExec := ctx.Value(util.ExecDetailsKey); stmtExec != nil {
+		detail := stmtExec.(*util.ExecDetails)
```

Comment:
Does it maybe nil in some scenario？

Reply:
No. If the value exists, it should not be nil, otherwise the context value should not exist.